icu4c/source/i18n/plurrule.cpp

```diff
@@ -1732,6 +1752,23 @@ int32_t FixedDecimal::getVisibleFractionDigitCount() const {
     return visibleDecimalDigitCount;
 }
 
+bool FixedDecimal::operator==(const FixedDecimal &other) const {
+    return source == other.source && visibleDecimalDigitCount == other.visibleDecimalDigitCount
+        && decimalDigits == other.decimalDigits && exponent == other.exponent;
+}
+
+UnicodeString FixedDecimal::toString() const {
+    char pattern[15];
+    char buffer[20];
+    if (exponent == 0) {
+        sprintf(pattern, "%.%df", visibleDecimalDigitCount);
+        sprintf(buffer, pattern, source);
+    } else {
+        sprintf(pattern, "%.%dfe%%d", visibleDecimalDigitCount);
+        sprintf(buffer, pattern, source, exponent);
+    }
+    return UnicodeString(buffer);
```

Comment:
Avoid implicit 8-bit to 16-bit conversion.  Please use `UnicodeString(buffer, -1, US_INV)`

Reply:
Done.